doc/man/pilot-getromtoken.1
.\"Generated by db2man.xsl. Don't modify this, modify the source.
.de Sh \" Subsection
.br
.if t .Sp
.ne 5
.PP
\fB\\$1\fR
.PP
..
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Ip \" List item
.br
.ie \\n(.$>=3 .ne \\$3
.el .ne 3
.IP "\\$1" \\$2
..
.TH "pilot-getromtoken" 1 "Copyright 1996-2007 FSF" "0.12.5" "PILOT-LINK"
.SH NAME
pilot-getromtoken \- Reads a ROM token from a Palm Handheld device.
.SH "SECTION"

.PP
pilot\-link: Tools

.SH "SYNOPSIS"

.PP
 \fIpilot\-getromtoken\fR [\fB\-p\fR|\fB\-\-port\fR <\fBport\fR>] [\fB\-\-version\fR] [\fB\-?\fR|\fB\-\-help\fR] [\fB\-\-usage\fR] [\fB\-q\fR|\fB\-\-quiet\fR] [\fB\-t\fR|\fB\-\-token\fR  \fBtoken\fR] [\fB\-s\fR|\fB\-\-sysinfo\fR] [\fB\-\-ram\fR] [\fB\-\-rom\fR] [\fIfilename\fR]

.SH "DESCRIPTION"

.PP
Tokens you may currently extract are:

.PP
adcc: Entropy for internal A\->D convertor calibration

.PP
irda: Present only on memory card w/IrDA support

.PP
snum: Device serial number (from Memory Card Flash ID)

.PP
 \fIpilot\-getromtoken\fR is used to fetch the ROM from your Palm handheld for use in debugging Palm applications through the use of POSE, the \fIPalm OS Emulator application\fR\&.

.PP
Note, because \fIpilot\-getrom\fR uses the low\-level RPC protocol to fetch the ROM image, and the new OS5 devices do not use this protocol, you currently cannot fetch an OS5 ROM image using this utility\&. This will be updated in a future release to handle fetching OS5 ROM images, using the debugger protocol\&.

.SH "OPTIONS"

.SS "pilot-getromtoken options"

                        \fB\-t\fR,
                        \fB\-\-token\fR \fBtoken\fR
                    
.PP
A ROM token to read (i\&.e\&. snum)

                        \fB\-s\fR, \fB\-\-sysinfo\fR
                    
.PP
Print SysInfo

                        \fB\-\-ram\fR
                    
.PP
Read RAM

                        \fB\-\-rom\fR
                    
.PP
Read ROM

                        <\fBfilename\fR>
                    
.PP
Write the ROM image to the file named by <\fIfilename\fR>\&. If this is omitted, the filename will be written out as the rom version name as follows: \fI 'pilot\-$VERSION\&.rom' \fR ,where an OS 3\&.5 image would be written as the filename of \fI'pilot\-3\&.5\&.0\&.rom'\fR\&.

.SS "Conduit Options"

                        \fB\-p\fR, \fB\-\-port\fR
                        <\fBport\fR>
                    
.PP
Use device file <\fIport\fR> to communicate with the Palm handheld\&. If this is not specified, \fIpilot\-getrom\fR will look for the \fI$PILOTPORT\fR environment variable\&. If neither are found or supplied, \fI pilot\-getrom \fR will print the usage information\&.

                        \fB\-q\fR, 
                        \fB\-\-quiet\fR
                    
.PP
Suppress 'Hit HotSync button' message

                        \fB\-v\fR, \fB\-\-version\fR
                    
.PP
Display version of \fIpilot\-getrom\fR and exit without connecting\&.

.SS "Help Options"

                        \fB\-h\fR, \fB\-\-help\fR
                    
.PP
Display the help synopsis for \fIpilot\-getrom\fR and exit without connecting\&.

                        \fB\-\-usage\fR 
                    
.PP
Display a brief usage message and exit without connecting\&.

.SH "USAGE"

.PP
Connect to a target Palm handheld and fetch the ROM for use in debugging Palm applications through the use of POSE, the \fIPalm OS Emulator\fR or with the \fICoPilot\fR application (deprecated)\&.

.SH "LICENSING"

.PP
 \fIWarning\fR: Please completely back up your Palm this program! The developers of \fI pilot\-link \fR are not responsible for any data loss that may occur from the use of this application\&.

.PP
Use of this program MAY place you in violation of your license agreement with Palmsource, Inc\&. or other OEM partners\&. Please read your Palm documentation (specifically the "Software License Agreement") before running this program\&.

.SH "AUTHOR"

.PP
 \fIpilot\-getrom\fR was originally written by Owen Stenseth\&.

.PP
This manual page was written by Neil Williams <linux@codehelp\&.co\&.uk> 

.SH "BUGS"

.SS "Known Bugs"

.PP
 \fIpilot\-getrom\fR does not work with OS5 devices as yet, due to changes in the architecture and design of the Palm HotSync protocol and file type changes from m68k to ARM\&. Other than that, \fIpilot\-getrom\fR has no known bugs at this time\&.

.SS "Reporting Bugs"

.PP
We have an online bug tracker\&. Using this is the only way to ensure that your bugs are recorded and that we can track them until they are resolved or closed\&. Reporting bugs via email, while easy, is not very useful in terms of accountability\&. Please point your browser to http://bugs\&.pilot\-link\&.org: \fIhttp://bugs.pilot-link.org\fR and report your bugs and issues there\&.

.SH "COPYRIGHT"

.PP
This program is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation; either version 2 of the License, or (at your option) any later version\&.

.PP
This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE\&. See the GNU General Public License for more details\&.

.PP
You should have received a copy of the GNU General Public License along with this program; if not, write to the Free Software Foundation, Inc\&., 51 Franklin St, Fifth Floor, Boston, MA 02110\-1301, USA\&.

.SH "SEE ALSO"

.PP
 \fIpilot\-getram\fR(1), \fIpilot\-link\fR(7)\&.



src/pilot-getromtoken.c
/*
 * $Id: pilot-getromtoken.c,v 1.19 2006/10/12 14:21:21 desrod Exp $ 
 *
 * pilot-getromtoken.c:  Retrieve a rom token from a device.
 *
 * Copyright (C) 2002, Owen Stenseth
 *
 * This program is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License as published by the
 * Free Software Foundation; either version 2 of the License, or (at your
 * option) any later version.
 *
 * This program is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
 * Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
 *
 */

#define DEFAULT_MODE mode_gettoken
#include "pilot-getrom.c"

